Formula/s/surelog.rb

```diff
@@ -0,0 +1,59 @@
+class Surelog < Formula
+  include Language::Python::Virtualenv
+
+  desc "SystemVerilog Pre-processor, parser, elaborator, UHDM compiler"
+  homepage "https://github.com/chipsalliance/Surelog"
+  url "https://github.com/chipsalliance/Surelog/archive/refs/tags/v1.74.tar.gz"
+  sha256 "0fb8d6e55ed3189ecb9aa3e33616d5b4b92f91fa25ff2e4415e1ee661d75fa88"
+  license "Apache-2.0"
+  head "https://github.com/chipsalliance/Surelog.git", branch: "master"
+
+  depends_on "antlr" => :build
+  depends_on "cmake" => :build
+  depends_on "nlohmann-json" => :build
+  depends_on "openjdk" => :build
+  depends_on "python@3.11" => :build
+  depends_on "six" => :build
+  depends_on "googletest" => :test
+  depends_on "pkg-config" => :test
+  depends_on "antlr4-cpp-runtime"
+  depends_on "capnp"
+  depends_on "uhdm"
+
+  resource "orderedmultidict" do
+    url "https://files.pythonhosted.org/packages/53/4e/3823a27d764bb8388711f4cb6f24e58453e92d6928f4163fdb01e3a3789f/orderedmultidict-1.0.1.tar.gz"
+    sha256 "04070bbb5e87291cc9bfa51df413677faf2141c73c61d2a5f7b26bea3cd882ad"
+  end
+
+  def install
+    venv = virtualenv_create(buildpath/"venv", "python3.11")
+    resources.each do |r|
+      venv.pip_install r
+    end
+
+    # Get antlr path
+    #
+    # Build shared library
+    system "cmake", "-S", ".", "-B", "build_shared",
+      "-DBUILD_SHARED_LIBS=ON",
+      "-DSURELOG_BUILD_TESTS=OFF",
+      "-DSURELOG_USE_HOST_GTEST=ON",
+      "-DSURELOG_USE_HOST_ANTLR=ON",
+      "-DSURELOG_USE_HOST_CAPNP=ON",
+      "-DSURELOG_USE_HOST_JSON=ON",
+      "-DSURELOG_USE_HOST_UHDM=ON",
+      "-DGTEST_LIBRARY=unused",
+      "-DGTEST_INCLUDE_DIR=unused",
+      "-DGTEST_MAIN_LIBRARY=unused",
+      "-DANTLR_JAR_LOCATION=#{Formula["antlr"].opt_prefix}/antlr-#{Formula["antlr"].version}-complete.jar",
+      "-DSURELOG_WITH_ZLIB=ON",
+      "-DCMAKE_INSTALL_RPATH=#{rpath}/surelog",
+      "-DPython3_EXECUTABLE=#{buildpath}/venv/bin/python", *std_cmake_args
+    system "cmake", "--build", "build_shared"
+    system "cmake", "--install", "build_shared"
+  end
+
+  test do
+    system bin/"surelog", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.